### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by cpopell** [st3v3nmw/obsidian-spaced-repetition#180] [Q&A]
How many flashcards do you expect this to scale to performantly?

**Comment by st3v3nmw** [comment]
I don't really have a number; I haven't come across performance issues on desktop (it takes about half a second to process the ≈ 1000 cards I have, distributed over some 120 md notes), it's much slower on mobile though (it takes almost 6 seconds for the same processing).

I'll implement some caching as mentioned in #171 which should drastically improve the processing times. Currently the plugin checks all the notes every time, could modify it to check only files that have been modified since the last review session.

**Reply by cpopell** [reply]
Got it! Thanks for the response. I'm up to fifty flashcards and I anticipate slowly growing towards 10, 20k over time.
